transforms.go

```diff
@@ -50,11 +50,21 @@ func buildBody(ctx context.Context, configuration configuration, diagnostic diag
 		person = &configuration.person
 	}
 	if person.Id != "" {
-		data["person"] = map[string]string{
+		personData := map[string]string{
 			"id":       person.Id,
 			"username": person.Username,
 			"email":    person.Email,
 		}
+		for key, value := range person.Extra {
+			// If the field on the extra map is already specified then skip it.
+			// This will prevent the extra map from overwriting fields like ID, Username or Email.
+			if _, ok := personData[key]; ok {
+				continue
```

Comment:
I would suggest removing 'continue' and checking for '!ok' 

Also some unit tests would be useful.

Reply:
I'm not gonna write tests for a code base that requires an API key. I'm also no longer at the company that was using this SDK.

If you wanna make these changes go for it; but I'll probably just close the PR. It's been more than 6 months.

Reply:
I'm not gonna write tests for a code base that requires an API key. I'm also no longer at the company that was using this SDK.

If you wanna make these changes go for it; but I'll probably just close the PR. It's been more than 6 months.